### DISCUSSION TITLE
Automated Traceback collection

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#646] [Feature Requests]
This might be a bit hard to do without any privacy issues, but it would be nice to collect tracebacks and general diagnostics from server operators.

The easiest way to make it work privacy-wise is to to make it a config option and ask when doing --copy-config

**Comment by godwhoa** [comment]
We can easily do it via. services like [sentry](https://sentry.io/for/python/), [bugsnag](https://www.bugsnag.com/platforms/python-error-reporting/) and dozens more.
Currently searching for self-hostable options.

**Comment by damienflament** [comment]
Which kind of data do you want to collect ?

If its just about crashes or performance, anonymising data seems easy: prevents collecting player names, IP, server name, map name etc.

**Comment by NotAFile** [comment]
I'm not really worried about any of that, since that data is public anyway. Anyone can join a server and get that information or just scrape it from the server list.
